## Q1:

#### The key factor that makes the difference between ideas that can, and cannot be examined and tested statistically is that whether we have data which can be examined to verify the idea.
#### I think the key criteria defining a good null hypothesis is that it should be able to test by data, and it shouldn't be too far from the actual value we get.
#### The difference between a null hypothesis and an alternative hypothesis is that the null hypothesis wants to show the case is true, while the alternative hypothesis wants to show the null hypothesis is wrong.

## Q2:

#### xi: It refers to the i^th data of the sample which is called X.
#### x bar: It refers to the mean of the sample.
#### mu: It refers to the population mean which is estimated by the mean of samples.
#### mu0: It refers to the population mean which is assumed in the null hypothesis.

## Q3:

#### Since p-value evaluates the probability of getting observed sequence when assuming the null hypothesis is true. Thus the precondition we must set is that we assume the null hypothesis is true.

## Q4:


#### When the p-value is samll, the sample where the test statistics we observed are in it is almost impossible to see in the population in the basis of the null hypothesis is true. However, the sample is randomly chosen, which means that it should not be so rare to find. Thus, it is more likely that the null hypothesis itself is ridiculous.

## Q5:

In [2]:
import numpy as np
import pandas as pd
import plotly.express as px

H0 = 0.5
observed_statistic = 0.645

np.random.seed(1)
n = 10000
n_size = 124
simulation_underH0 = np.zeros(n)

for i in range(n):
    random_improvement = np.random.choice([0,1],size = n_size, replace = True)
    simulation_underH0[i] = random_improvement.mean()

simulated_statistics = simulation_underH0
extreme_than_observed = abs(simulated_statistics - H0) >= abs(observed_statistic - H0)
extreme_than_observed.sum()/n

0.0015

Since p-value = 0.0015, we have strong evidence against the null hypothesis.

## Q6:

#### A small p-value cannot prove that the null hypothesis is false. It can only indicate that when we assume that the null hypothesis is true, the sample is very hard to be observed, which means that we may be more likely to believe that the null hypothesis is false but cannot ensure that the null hypothesis is false.
#### We cannot use p-value to prove that Fido is either innocent or guilty, no matter how high or low the p-value is.

## Q7:

In [11]:
import pandas as pd

patient_data = pd.DataFrame({
    "PatientID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Age": [45, 34, 29, 52, 37, 41, 33, 48, 26, 39],
    "Gender": ["M", "F", "M", "F", "M", "F", "M", "F", "M", "F"],
    "InitialHealthScore": [84, 78, 83, 81, 81, 80, 79, 85, 76, 83],
    "FinalHealthScore": [86, 86, 80, 86, 84, 86, 86, 82, 83, 84]
})

patient_data['HealthScoreChange'] = patient_data.FinalHealthScore-patient_data.InitialHealthScore

np.random.seed(1)
number_of_simulations = 10000
n_size = len(patient_data)
IncreaseProportionSimulations_underH0random = np.zeros(number_of_simulations)
for i in range(number_of_simulations):
    random_improvement = np.random.choice([0,1], size=len(patient_data), replace=True)
    IncreaseProportionSimulations_underH0random[i] = random_improvement.mean()

# Assuming H0 is that the improvement is random (50% chance)
population_parameter_value_under_H0 = 0.5

# Observed proportion of patients who improved
observed_statistic = (patient_data.HealthScoreChange > 0).mean()

# Simulated statistics
simulated_statistics = IncreaseProportionSimulations_underH0random

# One-sided test: Simulations "as or more extreme" for a greater-than test
SimStats_as_or_more_extreme_than_ObsStat = \
    simulated_statistics >= observed_statistic

# p-value calculation (one-sided)
p_value_one_sided = SimStats_as_or_more_extreme_than_ObsStat.sum() / number_of_simulations

# Output the results
print("Number of Simulations: ", number_of_simulations, "\n\n",
      "Number of simulated statistics (under H0)\n",
      'that are "as or more extreme" than the observed statistic: ',
      SimStats_as_or_more_extreme_than_ObsStat.sum(), "\n\n",
      'p-value\n(= simulations "as or more extreme" / total simulations): ', p_value_one_sided, sep="")


Number of Simulations: 10000

Number of simulated statistics (under H0)
that are "as or more extreme" than the observed statistic: 565

p-value
(= simulations "as or more extreme" / total simulations): 0.0565


The one-sided test is more specific, and typically has more power to detect an effect in the direction you're testing (in this case, improvement). However, it won't capture any cases where the observed statistic might be significantly lower than expected. By changing to a one-sided test, you're focusing only on the hypothesis that the patients' improvement is greater than what would be expected by chance, rather than testing for any difference (greater or less).

In practice, if the observed statistic falls in the direction of your one-tailed hypothesis, the p-value will generally be smaller for the one-tailed test because you're only focusing on one side of the distribution. However, if the observed statistic does not fall in the direction of your one-tailed hypothesis, the p-value may not necessarily be smaller (and could be larger).

Therefore, the p-value in a one-tailed test will usually be smaller than in a two-tailed test, as long as the result is in the direction you're testing.

#### Summary: Code Overview:

You provided a simulation-based hypothesis test to assess whether there is significant improvement in patient health scores, using a two-tailed (two-sided) test. The code ran 10,000 simulations to compare the observed proportion of patients with health score improvements to random chance (null hypothesis), and calculated the p-value.
One-Sided Test Adjustment:

We discussed how to adjust the code for a one-sided hypothesis test, where you’re only interested in whether the observed proportion of patients with health score improvements is greater than what would be expected by chance.
This involved modifying the condition for "as or more extreme" simulated statistics to focus only on one direction (greater-than) rather than both directions (greater or lesser).
Interpretation of Hypothesis Tests:

In a two-tailed test, deviations in both directions (higher or lower than the null hypothesis value) are considered, making the p-value larger since it accounts for both extremes.
In a one-tailed test, you only test deviations in one direction (greater than expected), so the p-value is typically smaller because it focuses on a single tail of the distribution.
Expected Difference in p-values:

For the same observed statistic, the p-value in a one-tailed test is usually smaller than in a two-tailed test because you're considering fewer extreme outcomes (only in one direction).
However, this is only true if the observed statistic falls in the direction you're testing (e.g., greater improvement). If not, the one-tailed p-value could be larger.

#### Link: https://chatgpt.com/share/670b064c-de6c-8003-ac55-7d6d43c096ef

## Q8:

Relationship: This experiment is based on the original experiment but with bigger sample.

Null Hypothesis: The probability of people can correctly tell the sequence of putting milk and pouring the tea is 0.5.

Alternative hypothesis: The probability of people can correctly tell the sequence of putting milk and pouring the tea is not 0.5.

In [18]:
import numpy as np
import pandas as pd
import plotly.express as px

H0 = 0.5
observed_statistic = 0.6125

np.random.seed(1)
n = 10000
n_size = 80
simulation_underH0 = np.zeros(n)

for i in range(n):
    random_improvement = np.random.choice([0,1],size = n_size, replace = True)
    simulation_underH0[i] = random_improvement.mean()

extreme_than_observed = abs(simulation_underH0 - H0) >= abs(observed_statistic - H0)
extreme_than_observed.sum()/n

0.0439

Since 0.01 < p-value = 0.0439 < 0.05, we have moderate evidence against the null hypothesis. Thus, the probability of people can correctly tell the sequence of putting milk and pouring the tea is not 0.5.

## Q9:

Somewhat